src/aleph/vm/orchestrator/views/authentication.py

```diff
@@ -89,12 +98,23 @@ def check_expiry(cls, values) -> dict[str, bytes]:
     @root_validator(pre=False, skip_on_failure=True)
     def check_signature(cls, values) -> dict[str, bytes]:
         """Check that the signature is valid"""
-        signature: bytes = values["signature"]
+        signature: list = values["signature"]
         payload: bytes = values["payload"]
         content = SignedPubKeyPayload.parse_raw(payload)
-        if not verify_wallet_signature(signature, payload.hex(), content.address):
-            msg = "Invalid signature"
-            raise ValueError(msg)
+
+        if content.chain == Chain.SOL:
+
+            try:
+                verify_signature(content.address, signature, payload.hex())
+            except BadSignatureError:
+                msg = "Invalid signature"
+                raise ValueError(msg)
+        elif content.chain == "ETH":
+            if not verify_wallet_signature(signature, payload.hex(), content.address):
```

Comment:
Maybe unify both methods for SOL and ETH into one unique method?
Also abstracting this common method and use one or another implementation depending on the chain will be a bit cleaner and easy to integrate with next chains.

Reply:
done